prog/base.rb

```diff
@@ -52,10 +57,17 @@ def self.label(label)
     (@labels ||= []) << label
 
     define_method :"hop_#{label}" do
+      incr_destroying if label == :destroy
       dynamic_hop label
```

Comment:
This adds an unnecessary conditional to every non-destroy hop.  You could avoid it by defining different methods depending on the conditional.  Up to you.

```suggestion
    if label == :destroy
      define_method :"hop_#{label}" do
        incr_destroying 
        dynamic_hop label
      end
     else
      define_method :"hop_#{label}" do
        dynamic_hop label
      end
```

Reply:
I like it. It adds a few lines, but it's better to optimize runtime and make the decision during loading.